### PR TITLE
Build GKv3 installer (DMG) for MacOS

### PR DESCRIPTION
**Pull request by hazzik**
Added a GitHub action that produces DMG installer containing unified (x64/arm64) build of the GEDKeeper3, that should run on both x64 (I've checked it works) and arm64 (Apple Silicon M1/M2 - I unfortunately do not have one, so could not check).

Because the DMG file are unsigned after download when it is tried to be opened it produces the "App Is Damaged and Can’t Be Opened" warning. See [here](https://www.howtogeek.com/803598/app-is-damaged-and-cant-be-opened/) for the information why this is happening and how to fix it.

TL;DR; Removed the `com.apple.quarantine` attribute after installation:

`xattr -d com.apple.quarantine /Applications/GEDKeeper3.app`

### CODE REVIEW

File: .github/workflows/package-mac.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 6.0.x
+    - name: Build
+      run: dotnet build -c MacOS_Debug -p:EnableDmgBuild=true
+      working-directory: projects/GKv3
+    - name: Upload DMG file
```
Comment (hazzik) [inline .github/workflows/package-mac.yml:23]: It crates an artifact that can be downloaded from Actions tab (Actions -> 'Package GEDKeeper 3 for MacOS' -> workflow run, at the bottom of 'Summary' page). Then it can be downloaded from there and distributed as a normal release file.

File: build_v3.mac.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+dotnet build projects/GKv3/GEDKeeper3.sln -c MacOS_Debug -p:MacBuildBundle=true
+
+# For details how to create a proper DMG HSF+ file see here: https://stackoverflow.com/a/7553878/259946.
+# The ISO9660/HFS is a supported format for DMG files and can be treated as uncompressed DMG.
+
+# sudo apt install mkisofs
```
Comment (hazzik) [inline build_v3.mac.sh:8]: `mkisofs`/`genisoimage` produces `ISO-9660` that is a valid DMG file that can be mounted on MacOS. See the link above how to make a compressed DMG file from ISO.

File: projects/GKv3/.run/GEDKeeper3.Mac.run.xml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="GEDKeeper3.Mac" type="DotNetProject" factoryName=".NET Project">
-    <option name="EXE_PATH" value="$PROJECT_DIR$/GEDKeeper3/bin/MacOS_Debug/net6.0/GEDKeeper3.app/Contents/MacOS/GEDKeeper3" />
```
Comment (hazzik) [inline projects/GKv3/.run/GEDKeeper3.Mac.run.xml:3]: This has changed because now we have a single target framework.

**Review by Serg-Norseman — APPROVED** [review]
👍 👍
